monkey/infection_monkey/control.py

```diff
@@ -55,12 +56,6 @@ def wakeup(self, parent=None):
         )
 
     def send_telemetry(self, telem_category, json_data: str):
-        if not self.server_address:
-            logger.error(
-                "Trying to send %s telemetry before current server is established, aborting."
-                % telem_category
-            )
-            return
```

Comment:
Are we sure we don't need this check anymore? We've only changed the type of `server_address`. I don't see how that affects the possibility of an attempt being made to send telemetry before the server was established. Unless this is stale code?

Reply:
Now that the address is a `SocketAddress` instead of a string, it will have a valid address. It may be useful to make `server_address` a readonly property as well, since I don't think we ever need to modify it after the `ControlClient` is created

Reply:
The check is valid nonetheless. Look at the code in `monkey.py`.

https://github.com/guardicore/monkey/blob/3accaccceb8670885e33e6168c37eb6f9a6e46d0/monkey/infection_monkey/monkey.py#L117-L123
https://github.com/guardicore/monkey/blob/3accaccceb8670885e33e6168c37eb6f9a6e46d0/monkey/infection_monkey/monkey.py#L153-L175

There's a possibility that `self._server_address` is `None` based on what `_connect_to_island_api()` returns.

Reply:
Ah. The type hints confused me. I still strongly encourage [this](https://github.com/guardicore/monkey/pull/2330#discussion_r977431883) in cases where exceptions are raised.